posthog/models/web_preaggregated/sql.py

```diff
@@ -1,17 +1,13 @@
-from django.conf import settings
-
 from posthog.clickhouse.table_engines import MergeTreeEngine, ReplicationScheme
+from posthog.clickhouse.cluster import ON_CLUSTER_CLAUSE
 from posthog.hogql.database.schema.web_analytics_s3 import get_s3_function_args
 
-CLICKHOUSE_CLUSTER = settings.CLICKHOUSE_CLUSTER
-CLICKHOUSE_DATABASE = settings.CLICKHOUSE_DATABASE
-
 
 def TABLE_TEMPLATE(table_name, columns, order_by):
     engine = MergeTreeEngine(table_name, replication_scheme=ReplicationScheme.REPLICATED)
 
     return f"""
-    CREATE TABLE IF NOT EXISTS {table_name}
+    CREATE TABLE IF NOT EXISTS {table_name} {ON_CLUSTER_CLAUSE(on_cluster=True)}
```

Comment:
The `ON_CLUSTER_CLAUSE` was required by the `test_async_migrations

Reply:
Thanks @orian !

Yeah, I had removed it for the migrations that were executed, those won't run again, but I didn't have to add it to the `clickhouse/schema.py` before but had to do it now to run the tests on those tables.

The thing is: there is a Regex on `clickhouse/schema.py` that expects it to exist: 
`get_table_name = lambda query: re.findall(r"[\.\s]`?([a-z0-9_]+)`?\s+ON CLUSTER", build_query(query))[0]`
https://github.com/PostHog/posthog/blob/b0a2c31779aa572cc4fd4b2a6401c53ff49677c9/posthog/clickhouse/schema.py#L259 and it is used on the [test_schema](https://github.com/PostHog/posthog/blob/8deaf4e8eaa7921b06052c2f1b4052b12c143ff1/posthog/clickhouse/test/test_schema.py#L15-L42) so I had to include it. 

Would you like me to try to make it more flexible for new tables?